libp2p/network/url_conversion_test.go

```diff
@@ -105,14 +104,35 @@ func TestPeerMultiAddrFromServiceURL_Error(t *testing.T) {
 			t.Run(testName, func(t *testing.T) {
 				serviceURL := fmt.Sprintf(testCase.serviceURLFormat, hostname)
 				actualMultiaddr, err := Libp2pMultiaddrFromServiceURL(serviceURL)
-				// TECHDEBT: assert specific errors
-				// Print resulting multiaddr to understand why no error.
-				require.ErrorContainsf(t, err, testCase.expectedErrContains, fmt.Sprintf("actualMultiaddr: %s", actualMultiaddr))
+				assert.Error(t, err, fmt.Sprintf("actualMultiaddr: %s", actualMultiaddr))
+				assert.True(t, strings.HasPrefix(err.Error(), testCase.expectedErr), fmt.Sprintf("expected error to start with %q, but got %q", testCase.expectedErr, err.Error()))
 			})
 		}
 	}
 }
 
+func ExampleLibp2pMultiaddrFromServiceURL() {
+	// Example: IPv4
+	peerMultiAddr, err := Libp2pMultiaddrFromServiceURL("www.google.com:8080")
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println(peerMultiAddr)
+
+	// Output: /ip4/142.251.40.164/tcp/8080
+}
+
+func ExampleLibp2pMultiaddrFromServiceURL_IPv6() {
```

Comment:
NIT: I think this warning will only appear when this line is modified in the future but if it takes less than 5 min to figure out what the proper `//nolint:govet` line would be to excuse this case I encourage doing so. Short of that, perhaps simply adding an extra `_` would stop it from complaining.